### PR TITLE
added storing of hex package during publishing

### PR DESCRIPTION
**Pull request by andpopov**
@Swizbiz Please review PR

### CODE REVIEW

File: src/main/java/com/artipie/hex/tarball/MetadataConfig.java [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+                            expr.setLength(0);
+                        } else {
+                            expr.append(line);
+                        }
+                    }
+                }
+                return new MetadataConfig(tuples);
+            }
+        } catch (IOException | ETP.ParseException e) {
+            throw new RuntimeException(e);
```
Comment (Swizbiz) [inline src/main/java/com/artipie/hex/tarball/MetadataConfig.java:109]: Please add information to exception message.

File: src/main/java/com/artipie/hex/tarball/HexPackageNameExtractor.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                        while((len = tar.read(buf)) != -1) {
+                            entryContent.write(buf, 0, len);
+                        }
+                        bytes = entryContent.toByteArray();
+                        break;
+                    }
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException(e);
```
Comment (Swizbiz) [inline src/main/java/com/artipie/hex/tarball/HexPackageNameExtractor.java:65]: Please add information to exception message.

File: src/main/java/com/artipie/hex/http/LocalHexSlice.java [hunk 3]
```diff
@@ -133,43 +135,51 @@ public Response response(final String line, final Iterable<Map.Entry<String, Str
...
-                            new ContentWithSize(body, headers)
+                    CompletableFuture
+                        .supplyAsync(() -> {
+                                byte[] bytes = asBytes(body);
+                                String tarName = HexPackageNameExtractor.extract(bytes).orElseThrow();
+                                Content content = new Content.From(bytes);
+                                try {
+                                    this.storage.save(
+                                        new Key.From(tarName),
+                                        content
```
Comment (Swizbiz) [inline src/main/java/com/artipie/hex/http/LocalHexSlice.java:148]: delete redundant local variable 'content' and collapse it to new Content.From(bytes)

File: pom.xml [hunk 4]
```diff
@@ -127,5 +137,15 @@ SOFTWARE.
...
     </dependency>
+    <dependency>
+      <groupId>com.github.metadave</groupId>
+      <artifactId>etp</artifactId>
+      <version>0.6-SNAPSHOT</version>
+    </dependency>
+    <dependency>
+      <groupId>org.apache.commons</groupId>
+      <artifactId>commons-compress</artifactId>
+      <version>1.21</version>
```
Comment (Swizbiz) [inline pom.xml:148]: extract version to property

File: pom.xml [hunk 5]
```diff
@@ -127,5 +137,15 @@ SOFTWARE.
       <artifactId>protobuf-java</artifactId>
       <version>${protobuf-version}</version>
     </dependency>
+    <dependency>
+      <groupId>com.github.metadave</groupId>
+      <artifactId>etp</artifactId>
+      <version>0.6-SNAPSHOT</version>
```
Comment (Swizbiz) [inline pom.xml:143]: extract version to property

File: pom.xml [hunk 6]
```diff
@@ -78,6 +78,16 @@ SOFTWARE.
     <testcontainers.version>1.17.3</testcontainers.version>
     <vertx.server.version>1.0.0</vertx.server.version>
   </properties>
+  <repositories>
```
Comment (Swizbiz) [inline pom.xml:81]: maven central contain version 0.4 https://search.maven.org/artifact/com.github.metadave/etp/0.4/jar

Do you see deference between 0.4 and 0.6-SNAPSHOT version?

**Review by Swizbiz — CHANGES_REQUESTED** [review]
Thanks for PR. 
Please check my comments

File: pom.xml (hunk 4 above)
Comment (andpopov) [reply pom.xml]: @Swizbiz Thanks for your comments
Please review cahnges

**Review by Swizbiz — APPROVED** [review]
thanks

**Pull request merged**